### PR TITLE
275 web google maps in configuration process

### LINKED ISSUE #275: [WEB] Google maps in configuration process
- [x] Add a map during the configuration process
- [x] Display a marker after a click on this map
- [x] Save the latitude and longitude in the Container object
- [x] The user is free to add it anywhere, we don't process the input to validate it
- [x] Get user location to set initial display of maps

### PR DESCRIPTION
**Pull request by CedricCORGE**
close #275 

Capture d’écran du 2024-04-08 20-05-29
Capture d’écran du 2024-04-08 21-22-11
image

A permanant marker is on the center to determine the position (no other possibility).

### CODE REVIEW

**Review by tbellicha — APPROVED** [review]

File: Configurateur/front/lib/screens/container-creation/design_screen.dart [hunk 1]
```diff
@@ -333,6 +341,8 @@ class DesignScreenState extends State<DesignScreen> {
 
   void goNext() async {
     if (widget.id == null) {
+      debugPrint('width: ${widget.width}');
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen.dart:344]: To remove

File: Configurateur/front/lib/screens/container-creation/design_screen.dart [hunk 2]
```diff
@@ -121,6 +125,9 @@ class DesignScreenState extends State<DesignScreen> {
     dynamic decode = jsonDecode(widget.lockers!);
 
     for (int i = 0; i < decode.length; i++) {
+      if (decode[i]['type'] == 'Design personnalisé') {
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen.dart:128]: Could be better but ok

File: Configurateur/front/lib/screens/container-creation/maps_screen.dart [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  State<MapsScreen> createState() => MapsState();
+}
+
+class MapsState extends State<MapsScreen> {
+  String jwtToken = '';
+  final Completer<GoogleMapController> _controller =
+      Completer<GoogleMapController>();
+
+  static CameraPosition _kGooglePlex = const CameraPosition(
+    target: LatLng(47.210528, -1.566956),
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/maps_screen.dart:39]: Where is it ? Just add a comment for comprehension

**Review by 31Nathan — CHANGES_REQUESTED** [review]
You forgot the doc of the create-ctn route and the schema need to be updated in the doc

File: Configurateur/front/lib/screens/container-creation/maps_screen.dart (hunk 3 above)
Comment (CedricCORGE) [reply Configurateur/front/lib/screens/container-creation/maps_screen.dart:39]: epitech nantes

**Review by 31Nathan — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**